modules/node/src/channel/channel.service.spec.ts

```diff
@@ -0,0 +1,129 @@
+import { mkHash, mkXpub } from "../test/utils";
+import { CFCoreTypes, AppInstanceJson } from "../util";
+
+import { Channel } from "./channel.entity";
+import { ChannelService } from "./channel.service";
+import { ChannelRepository } from "./channel.repository";
+import { mkAddress } from "../../dist/src/test/utils";
+import { ConfigService } from "../config/config.service";
+import { OnchainTransaction } from "../onchainTransactions/onchainTransaction.entity";
+import { OnchainTransactionRepository } from "../onchainTransactions/onchainTransaction.repository";
+import { BigNumber } from "ethers/utils";
+import { DefaultApp, OutcomeType } from "@connext/types";
+
+class MockCFCoreService {
+  cfCore = {
+    freeBalanceAddress: mkAddress("0xabcdef"),
+  };
+
+  async deposit(): Promise<CFCoreTypes.DepositResult> {
+    console.log("Called mock function deposit()");
+    return {
+      multisigBalance: One,
+      tokenAddress: AddressZero,
+      transactionHash: mkHash("0xa"),
+    };
+  }
+
+  async proposeAndWaitForAccepted(): Promise<CFCoreTypes.ProposeInstallResult> {
+    return {
+      appInstanceId: mkHash("0xabc"),
+    };
+  }
+
+  async getCoinBalanceRefundApp(): Promise<AppInstanceJson | undefined> {
+    return undefined;
+  }
+}
+
+class MockChannelRepository extends ChannelRepository {
+  async findByMultisigAddress(): Promise<Channel | undefined> {
+    const channel = new Channel();
+    channel.available = true;
+    channel.collateralizationInFlight = false;
+    channel.multisigAddress = mkAddress("0xAAA");
+    channel.nodePublicIdentifier = mkXpub("xpubAAA");
+    channel.userPublicIdentifier = mkXpub("xpubBBB");
+    channel.id = 1;
+    return channel;
+  }
+}
+
+class MockOnchainTransactionRepository extends OnchainTransactionRepository {
+  async addCollateralization(): Promise<OnchainTransaction> {
+    return new OnchainTransaction();
+  }
+}
+
+class MockEthProvider {
+  async getTransaction(): Promise<TransactionResponse> {
+    return {
+      blockHash: mkHash(),
+      blockNumber: 1,
+      chainId: 1337,
+      confirmations: 10,
+      data: "0x",
+      from: mkAddress("0xabc"),
+      gasLimit: One,
+      gasPrice: One,
+      hash: mkHash("0xbbb"),
+      nonce: 1,
+      r: "foo",
+      s: "bar",
+      timestamp: 111,
+      to: mkAddress("0xdef"),
+      v: 1,
+      value: Zero,
+    } as any;
+  }
+
+  async getBalance(): Promise<BigNumber> {
+    return One;
+  }
+}
+
+class MockConfigService {
+  getEthProvider(): JsonRpcProvider {
+    return new MockEthProvider() as any;
+  }
+
+  async getDefaultAppByName(): Promise<DefaultApp> {
+    return {
+      actionEncoding: "",
+      allowNodeInstall: true,
+      appDefinitionAddress: mkAddress("0xa"),
+      chainId: 1337,
+      name: "SimpleLinkedTransferApp",
+      outcomeType: OutcomeType.SINGLE_ASSET_TWO_PARTY_COIN_TRANSFER,
+      stateEncoding: "",
+    };
+  }
+}
+
+describe.skip("Channel Service", () => {
```

Comment:
this should be unskipped before merging

Reply:
This isn't a real test, I was playing around with the test framework a bit.